### PR TITLE
Bug 1137164 - Add a shebang to setup.py

### PR DESCRIPTION
**Pull request by tojonmz**
This fixes Bugzilla bug [1137164](https://bugzilla.mozilla.org/show_bug.cgi?id=1137164).

In addition to the docs I also modified Dockerfile, but didn't touch .travis.yml, treeherder.pp or update.py. Not sure if that is correct, but my initial best guess.

Caveat I haven't tested all those related changes, so I defer to @edmorley  on his wisdom :)

Some additional trailing white space removed from installation.rst.

Adding Ed for review.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -24,12 +24,12 @@ Setting up Vagrant
      >vagrant up
 
   Note: If you encounter an error saying "It appears your machine doesn't support NFS, or there is not an
-  adapter to enable NFS on this machine for Vagrant.", then you need to install ``nfs-kernel-server`` using the commnad: 
+  adapter to enable NFS on this machine for Vagrant.", then you need to install ``nfs-kernel-server`` using the commnad:
```
Comment (edmorley) [inline docs/installation.rst]: Whilst you are here, s/commnad/command/ :-)

Comment (tojonmz) [inline docs/installation.rst]: Good one :) Will do.

**Comment by edmorley** [comment]
I think also we should refer to |./setup.py| since not all systems have '.' on the PATH.

**Comment by tojonmz** [comment]
Agreed.

**Pull request merged**